File: solutions/elixir/bob/1/bob.exs
defmodule Bob do
  def hey(input) do
    cond do
      is_blank? input -> "Fine. Be that way!"
      is_question?(input) and is_caps?(input) and not just_numbers?(input) -> "Calm down, I know what I'm doing!"
      is_question?(input) -> "Sure."
      is_caps?(input) and not just_numbers?(input) -> "Whoa, chill out!"
      true -> "Whatever."
    end
  end

  defp is_question?(input) do
    String.ends_with?(input, "?")
  end

  defp is_caps?(input) do
    input == String.upcase(input)
  end

  defp is_blank?(input) do
    "" == String.trim(input)
  end

  defp just_numbers?(input) do
    pattern = :binary.compile_pattern([" ", ",", "?", ";", "."])
    String.split(input, pattern, trim: true)
      |> Enum.join()
      |> String.match?(~r/^\d+$/)
  end
end

File: solutions/python/bob/2/bob.py
"""Module for interacting with a prompt"""

def response(hey_bob: str) -> str:
    """
    Respond to a prompt as Bob.

    :param hey_bob: str - The prompt
    :return: str - Bob's response
    """
    
    processed = hey_bob.strip()

    is_question = processed.endswith("?")
    is_shouting = processed.isupper()
    is_silence = not processed

    if is_shouting and is_question:
        return "Calm down, I know what I'm doing!"
    if is_question:
        return "Sure."
    if is_shouting:
        return "Whoa, chill out!"
    if is_silence:
        return "Fine. Be that way!"

    return "Whatever."
